### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by okbel**
# Auth

## What's new with this PR

- Adds Coral-UI
    - Dialog
    - Button
    - Success UI Comp
    - Super Cool Coral Spinner (Yes, Coral Color :heart_eyes:)
- Sign In
    - Sign In with Facebook
    - Sign In as a Local User
- Sign Up
    - Email Availability before sending the Full Sign Up Form (Feature)
- Forgot Pasword
- User Box
    - Supplies info from the user allowing him to log out.
- Logout
- ES/EN i18n Support
- Redux Support
- Availability and User creation endpoint

It should look like this
out

### CODE REVIEW

**Comment by okbel** [comment]
@wyattjoh @impronunciable Can you take a look at this PR? :) Let me know what you think!

File: redis.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+const redis = require('redis');
+const debug = require('debug')('talk:redis');
+const url = process.env.TALK_REDIS_URL || 'redis://localhost';
+
+const client = redis.createClient(url, {
+  retry_strategy: function(options) {
+    if (options.error.code === 'ECONNREFUSED') {
```
Comment (wyattjoh) [inline redis.js]: Change this line to:

```
if (options.error && options.error.code === 'ECONNREFUSED') {
```

File: redis.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    // reconnect after
+    return Math.max(options.attempt * 100, 3000);
+  }
+});
+
+// client.on('error', (err) => {
+//   throw err;
+// });
```
Comment (wyattjoh) [inline redis.js]: Delete this commented out code.

**Review by wyattjoh — CHANGES_REQUESTED** [review]

File: client/coral-framework/helpers/error.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export default {
+  email: 'Not a valid E-Mail',
+  password: 'Password must be at least 8 characters',
```
Comment (gabelula) [inline client/coral-framework/helpers/error.js]: we are not translating these?

File: client/coral-plugin-commentbox/CommentBox.js [hunk 4]
```diff
@@ -51,18 +52,9 @@ class CommentBox extends Component {
   }
 
   render () {
-    const {styles, reply} = this.props;
+    const {styles, reply, canPost} = this.props;
     // How to handle language in plugins? Should we have a dependency on our central translation file?
```
Comment (gabelula) [inline client/coral-plugin-commentbox/CommentBox.js:56]: let's remove this comment.

**Review by gabelula — CHANGES_REQUESTED** [review]

**Review by gabelula — APPROVED** [review]

**Comment by gabelula** [comment]
There is still a file that miss the change from i18n to modules.  The file `client/coral-admin/src/containers/Community/NoResults.js`

**Review by wyattjoh — APPROVED** [review]

**Comment by okbel** [comment]
@wyattjoh @gabelula All changes applied. Thank you both. Email Availability has been removed :(
Feel free to test and merge! 👍

**Review by gabelula — APPROVED** [review]

**Pull request merged**